### ISSUE TITLE
training time become longer

### ISSUE THREAD

**Issue by xsr-ai** [davidsandberg/facenet#657]
my environment info：
tensorflow：r1.6
GPU：gtx 1060 6GB
facenet backend：Inception_resnet_v1
computer RAM：16GB
datasets：
Total number of classes: 46221
Total number of examples: 3669400
batch size：60
epoch size：1000

before training，i change nrof_preprocess_threads to 2 which original value is 4, or change data_flow_ops.FIFOQueue capacity to 600000 which original value is 100000, that all can be let one epoch training time go faster at begin, but when training continue, the training time become lower.

if change nrof_preprocess_threads to 2, training time show below:
Epoch: [0][1/1000]	Time 7.075	Loss 4.254	RegLoss 2.636
Epoch: [0][2/1000]	Time 5.576	Loss 4.174	RegLoss 2.636
Epoch: [0][3/1000]	Time 0.410	Loss 4.437	RegLoss 2.636
Epoch: [0][4/1000]	Time 0.398	Loss 4.532	RegLoss 2.636
Epoch: [0][5/1000]	Time 0.412	Loss 4.349	RegLoss 2.636
Epoch: [0][6/1000]	Time 0.414	Loss 4.502	RegLoss 2.636
Epoch: [0][7/1000]	Time 0.382	Loss 4.232	RegLoss 2.636
Epoch: [0][8/1000]	Time 0.408	Loss 4.345	RegLoss 2.636
Epoch: [0][9/1000]	Time 0.404	Loss 4.329	RegLoss 2.636
Epoch: [0][10/1000]	Time 0.411	Loss 4.185	RegLoss 2.636
Epoch: [0][11/1000]	Time 0.455	Loss 4.210	RegLoss 2.636
Epoch: [0][12/1000]	Time 0.451	Loss 4.115	RegLoss 2.636
Epoch: [0][13/1000]	Time 0.429	Loss 4.095	RegLoss 2.636
Epoch: [0][14/1000]	Time 0.462	Loss 3.829	RegLoss 2.636
Epoch: [0][15/1000]	Time 0.421	Loss 4.272	RegLoss 2.636
Epoch: [0][16/1000]	Time 0.425	Loss 4.510	RegLoss 2.636
Epoch: [0][17/1000]	Time 0.456	Loss 4.068	RegLoss 2.636
Epoch: [0][18/1000]	Time 0.426	Loss 4.162	RegLoss 2.636
Epoch: [0][19/1000]	Time 0.438	Loss 4.291	RegLoss 2.636
Epoch: [0][20/1000]	Time 0.457	Loss 4.315	RegLoss 2.636
Epoch: [0][21/1000]	Time 0.426	Loss 4.145	RegLoss 2.636
Epoch: [0][22/1000]	Time 0.423	Loss 4.007	RegLoss 2.636
Epoch: [0][23/1000]	Time 0.427	Loss 4.222	RegLoss 2.636
Epoch: [0][24/1000]	Time 0.458	Loss 4.347	RegLoss 2.636
Epoch: [0][25/1000]	Time 0.459	Loss 4.113	RegLoss 2.636
Epoch: [0][26/1000]	Time 0.458	Loss 4.545	RegLoss 2.636
Epoch: [0][27/1000]	Time 0.429	Loss 4.217	RegLoss 2.636
Epoch: [0][28/1000]	Time 0.455	Loss 3.968	RegLoss 2.636
Epoch: [0][29/1000]	Time 0.454	Loss 4.514	RegLoss 2.636
Epoch: [0][30/1000]	Time 0.458	Loss 3.950	RegLoss 2.636
Epoch: [0][31/1000]	Time 0.428	Loss 4.428	RegLoss 2.636
Epoch: [0][32/1000]	Time 0.425	Loss 4.518	RegLoss 2.636
Epoch: [0][33/1000]	Time 0.453	Loss 4.178	RegLoss 2.636
Epoch: [0][34/1000]	Time 0.450	Loss 4.231	RegLoss 2.636
Epoch: [0][35/1000]	Time 0.455	Loss 4.218	RegLoss 2.636
Epoch: [0][36/1000]	Time 0.431	Loss 4.181	RegLoss 2.636
Epoch: [0][37/1000]	Time 0.458	Loss 4.254	RegLoss 2.636
Epoch: [0][38/1000]	Time 0.455	Loss 4.225	RegLoss 2.636
Epoch: [0][39/1000]	Time 1.672	Loss 4.381	RegLoss 2.636
Epoch: [0][40/1000]	Time 1.557	Loss 4.243	RegLoss 2.636
Epoch: [0][41/1000]	Time 1.552	Loss 4.080	RegLoss 2.636
Epoch: [0][42/1000]	Time 1.441	Loss 4.147	RegLoss 2.636
Epoch: [0][43/1000]	Time 1.479	Loss 4.496	RegLoss 2.636
Epoch: [0][44/1000]	Time 1.413	Loss 4.283	RegLoss 2.636
Epoch: [0][45/1000]	Time 1.575	Loss 4.547	RegLoss 2.636
Epoch: [0][46/1000]	Time 1.396	Loss 3.918	RegLoss 2.636
Epoch: [0][47/1000]	Time 1.695	Loss 4.230	RegLoss 2.636
Epoch: [0][48/1000]	Time 1.557	Loss 4.051	RegLoss 2.636
Epoch: [0][49/1000]	Time 1.475	Loss 4.288	RegLoss 2.636
Epoch: [0][50/1000]	Time 1.524	Loss 3.997	RegLoss 2.636

if change data_flow_ops.FIFOQueue capacity to 600000, training time show below:
Epoch: [0][278/1000]	Time 0.437	Loss 4.531	RegLoss 2.637
Epoch: [0][279/1000]	Time 0.461	Loss 4.181	RegLoss 2.637
Epoch: [0][280/1000]	Time 0.444	Loss 4.281	RegLoss 2.637
Epoch: [0][281/1000]	Time 0.440	Loss 4.428	RegLoss 2.637
Epoch: [0][282/1000]	Time 0.465	Loss 4.250	RegLoss 2.637
Epoch: [0][283/1000]	Time 0.441	Loss 4.269	RegLoss 2.637
Epoch: [0][284/1000]	Time 0.454	Loss 4.350	RegLoss 2.637
Epoch: [0][285/1000]	Time 0.464	Loss 4.492	RegLoss 2.637
Epoch: [0][286/1000]	Time 1.670	Loss 4.431	RegLoss 2.637
Epoch: [0][287/1000]	Time 1.390	Loss 4.406	RegLoss 2.637
Epoch: [0][288/1000]	Time 1.485	Loss 4.427	RegLoss 2.637
Epoch: [0][289/1000]	Time 1.561	Loss 4.440	RegLoss 2.637
Epoch: [0][290/1000]	Time 1.411	Loss 4.280	RegLoss 2.637
Epoch: [0][291/1000]	Time 1.395	Loss 4.560	RegLoss 2.637
Epoch: [0][292/1000]	Time 1.400	Loss 4.261	RegLoss 2.637
Epoch: [0][293/1000]	Time 1.470	Loss 4.385	RegLoss 2.637
Epoch: [0][294/1000]	Time 1.342	Loss 4.529	RegLoss 2.637
Epoch: [0][295/1000]	Time 1.503	Loss 4.558	RegLoss 2.637
Epoch: [0][296/1000]	Time 1.407	Loss 4.189	RegLoss 2.637
Epoch: [0][297/1000]	Time 1.478	Loss 4.415	RegLoss 2.637
Epoch: [0][298/1000]	Time 1.488	Loss 4.361	RegLoss 2.637

why the one epoch time aways become long? i had try to adjust the queue enqueue、dequeue and it's capacity but only delay the speed of training time become longer, i think the source code of data feeding maybe have undiscovered bug.

best regards.

**Comment by VincentVon** [comment]
I saved the training images on my .m2 ssd, and it's running very fast.
When I save  these images on hdd, it's takes 3 or 4 times time.
My advice is save these images on one ssd, and check again.

**Comment by xsr-ai** [comment]
but my dataset is large, so it can't save at ssd, and hard disk is not the sufficient reason，because begin the training time is faster. btw, i had done a experiment, i converted the image to tfrecord, then can sulove this issue.
